services/core/events/blockchain.js

```diff
@@ -49,14 +49,19 @@ module.exports = [
 						util.inspect(data),
 					].join('\n'));
 				}
-			});
+			};
+			if (!Signals.get('newBlock').has(newBlockListener)) {
+				logger.debug('newBlock has newBlockListener: ', Signals.get('newBlock').has(newBlockListener));
+				Signals.get('newBlock').add(newBlockListener);
+				logger.debug('newBlock has now registered newBlockListener: ', Signals.get('newBlock').has(newBlockListener));
+			}
```

Comment:
Is there any reason to keep that logic here instead of implementing in `Signal.get()`?

Please update it if possible.

Reply:
Done!